sql/moz-fx-data-shared-prod/telemetry_dev_cycle_derived/glean_metrics_stats_v1/query.sql

```diff
@@ -0,0 +1,50 @@
+WITH glean_app_with_parsed_expiry_date AS (
+  SELECT
+    glean.glean_app,
+    glean.metric,
+    glean.type,
+    glean.first_seen_date AS release_date,
+    glean.last_seen_date AS last_date,
+    glean.expires,
+    CASE
+      WHEN glean.expires = "never"
+        THEN NULL
+      WHEN REGEXP_CONTAINS(glean.expires, r'[0-9]{4}-[0-9]{2}-[0-9]{2}')
+        THEN CAST(glean.expires AS DATE)
+      ELSE releases.publish_date
+    END AS expiry_date,
+  FROM
+    `telemetry_dev_cycle_external.glean_metrics_stats_v1` AS glean
+  LEFT JOIN
+    `telemetry_dev_cycle_derived.firefox_major_release_dates_v1` AS releases
+  ON
+    glean.expires = CAST(releases.version AS STRING)
+    AND COALESCE(
+      REGEXP_EXTRACT(glean.glean_app, r"beta"),
```

Comment:
Is this just checking if it contains a specific substring?

Reply:
no this is in the join 
so if the `expires` is a version number, i need to join it with the releases table on the version and the channel. if there is a beta in the glean_app, I'm joining it to beta channel, if there is a "nightly" to nightly and else to release